packages/components/src/custom-select-control/test/index.js

```diff
@@ -87,4 +88,127 @@ describe( 'CustomSelectControl', () => {
 			screen.getByRole( 'button', { name: 'Custom select' } )
 		).toHaveTextContent( 'Hint' );
 	} );
+
+	it( 'Initial value should be replaced when a new item is selected', async () => {
+		const user = userEvent.setup();
+		const mock = jest.fn();
+
+		render(
+			<CustomSelectControl
+				onChange={ ( { selectedItem } ) => mock( selectedItem ) }
+				options={ options }
+				__nextUnconstrainedWidth
+			/>
+		);
+
+		const currentValue = screen.getByRole( 'button' );
+
+		expect( currentValue ).toHaveTextContent( /violets/i );
```

Comment:
When testing select controls, in addition to testing the displayed text content, we should also test the `value` of the underlying selected item. Think about it like an old-school `select` field with a few `option`s inside it. Each option has a different `value` and a different text content. Making sure both the text content and the actual value are both valid is especially valuable when we are building custom components that simulate the HTML select behavior.

Reply:
This may be tricky with the current (legacy) `CustomSelectControl`, as it doesn't look like the component exposes in the DOM any semantics related to the selected `value` apart from the text contents of the `button` 😅 

I also noticed that, when opening the popover and showing the `listbox` with all the `options`, the `aria-selected` attribute is not assigned correctly to the selected option, but is instead assigned to whatever item is hovered/focused 🙈 

Not sure what's the best thins to do here: — leave the test as-is, or tentatively add a portion of the test that checks for the expected behaviour which will fail on this PR

Reply:
> tentatively add a portion of the test that checks for the expected behaviour which will fail on this PR

This is a good idea IMHO as it demonstrates the superiority of the new version of the component.

Reply:
Definitely! In the PR description I have a section with ideas for future tests and the first one is in regards to this: 

>After making changes to the component, the following tests/improvements could be implemented:
>Replace `toHaveTextContent` with `toHaveValue`

I wanted to explicitly point it out as I knew it would be disappointing in its current state. 😄